Context root doesn't include a random string

OndroMih opened the pull request:
For apps deployed via Admin Console, the uploaded file was copied to a temp file,  which contained a random string. Now, the temp directory has a random string,  but the file in it has the same name.
Context root is generated from the file name and doesn't include any random string with this commit.

Before, when an app was deployed with default values:
![image](https://github.com/eclipse-ee4j/glassfish/assets/2195988/b1273f49-e11b-4642-b243-0ddcdbf79bbc)

Now:
![image](https://github.com/eclipse-ee4j/glassfish/assets/2195988/1da375b3-3d02-4632-99aa-1f46cfef3bfd)

OndroMih committed:
Context root doesn't include a random string
For apps deployed via Admin Console, the uploaded file was copied to a temp file, 
which contained a random string. Now, the temp directory has a random string, 
but the file in it has the same name.
Context root is generated from the file name and doesn't include 
a random string with this commit.
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

dmatej commented on appserver/admingui/core/src/main/java/org/glassfish/admingui/handlers/WoodstockHandler.java:
```diff
@@ -193,6 +185,16 @@ public static void uploadFileToTempDir(HandlerContext handlerCtx) {
 
+    private static File createTempFileWithOriginalFileName(String name) throws IOException {
+        final Path dir = Files.createTempDirectory(name);
+        dir.toFile().deleteOnExit();
```
Shouldn't we delete the directory too? 
Those deleteOnExit usages will be reported by SonarQube because deleteOnExit is done just by shutdown hooks and is not something to rely on. Maybe we should target it later. Also ignoring file.delete() result can be reported by SpotBugs (which is not your code, it was already there).

- https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-2308/?search=deleteOn

OndroMih replied:
I updated the code:
* directory is also removed in the `deleteFileFromTempDir` handler
* `Files.deleteIfExists` used instead of `file.delete()`, shouldn't be reported by SpotBugs
* On IO exception, the temp files are removed if possible and an error is reported in GUI - I don't know why, but before it continued with a file, which might have been corrupted or empty

OndroMih committed:
Context root doesn't include a random string
- also remove after file is uploaded
- improve file handling for uploaded files
- give an error in GUI if IOException
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

dmatej commented on appserver/admingui/core/src/main/java/org/glassfish/admingui/handlers/WoodstockHandler.java:
```diff
@@ -178,11 +171,13 @@ public static void uploadFileToTempDir(HandlerContext handlerCtx) {
                     }
                 } catch (Exception ex) {
-                    //Handle AbsolutePathException here
+                    // ignore nested exception, handle the original exception only
```
You can use ioex.addSuppressed here.
